### PR TITLE
Avoid reading the property if hasOwnProperty is false

### PR DESCRIPTION
**Pull request by sebmarkbage**
Easy with an Object.is polyfill

### CODE REVIEW

**Comment by spicyj** [comment]
uh, accepted.

**Comment by sebmarkbage** [comment]
thanks :)

I needed it in because I think this might be the RC if everything looks good.

File: src/classic/element/ReactElementValidator.js [hunk 1]
```diff
@@ -292,6 +292,19 @@ function warnForPropsMutation(propName, element) {
   );
 }
 
+// Inline Object.is polyfill
+function is(a, b) {
+  if (a !== a) {
+    // NaN
+    return b !== b;
```
Comment (Pouja) [inline src/classic/element/ReactElementValidator.js]: Just of curiosity why are you not just returning false?

Comment (sebmarkbage) [inline src/classic/element/ReactElementValidator.js]: They're equal if b is also NaN.

Comment (Pouja) [inline src/classic/element/ReactElementValidator.js]: Ah TIL thank you!

**Pull request merged**